### ISSUE TITLE
[ETHEREUM-CONTRACTS] The Giga Refactoring (Tests)

### ISSUE THREAD

**Issue by 0xdavinchee** [superfluid-finance/protocol-monorepo#1036]
## Reasoning
The SF test suite has grown in size and will likely continue to grow as we add new features and move forward. In addition, the bulk of the time spent doing smart contract development is not the implementation, but rather it is creating a spec prior to implementation and writing tests after. Given the inordinate amount of time spent testing, it makes sense to make the experience of writing tests as comfortable, foolproof and efficient as possible. All of this becomes exponentially more important as we continue to scale.

### Developer Experience 
- [x] Break down refactoring into more digestible phases instead of one gigantic refactor
- [x] 100% Typescript (#1069)
- [ ] Restructure organization of folders/files for tests
- [ ] Usage of signers as accounts
- [ ] Remove js-sdk and decentral.ee dependencies => move to using hardhat toolbox entirely
- [ ] Separation of integration/unit/fuzzing tests
- [ ] Have a clear architecture for the refactored testing framework
- [ ] Have some documentation for the refactored testing framework
- [ ] Review all test cases: remove redundant tests and move tests to foundry
- [ ] https://github.com/superfluid-finance/protocol-monorepo/issues/534 as part of the refactor
- [ ] Mainnet forking via hardhat/foundry added
- [ ] Instructions on tenderly-cli debugging added (instructions are known, but blocked by hardhat/tenderly being broken after v2.9.4)
- [ ] ~hardhat-vscode debugging set up~

### Nice to haves
- [ ] Hardhat compilation pipeline
- [ ] Use pnpm instead of yarn - allows for much faster local installation (about the same on github actions)

### Breakdown (Stages)
I think a good way to break this down is to go from a high-level (macro) and then get down to the minutiae and details.
1. Move to 100% TypeScript - the reason we start with this even though this is the hardest task and a fairly giga refactoring is that it will make the remaining refactoring **THAT** much more comfy
2. Rearchitect the testing framework and the organization of folders/test files and the testing process in general
3. Refactoring of test environment - accounts should be signers, apply consistency in terms of functions taking in one type, not name sometimes and addresses other times, etc.

**Comment by hellwolf** [comment]
please update this @0xdavinchee

**Comment by 0xdavinchee** [comment]
most of the points in this issue are no longer relevant, the plan is to keep our existing JS tests as is and to just write any new tests in Foundry
there are no plans to migrate existing JS tests over to Foundry.

**Comment by hellwolf** [comment]
lovely :)
